interview-questions.md

```diff
@@ -8,47 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is an instance variable in Ruby? How is it different from other variables?
 
   Your answer:
+  Instance variables are variables that are created using the @ symbol. They are able to store data. For example, a car can have different attributes such as @oil or @wheels. These are instance variables that hold information. An instance variable is locally scoped to the class it belongs to.
 
   Researched answer:
-
+  An instance variable is a kind of variable that starts with the @ symbol & is used inside classes to give them a place to store data. 
+  
+  Things to note:
+  -An undefined instance variable always returns nil
+  -Instance variables don’t have to be created in the initialize method, but that’s how you give them an initial value
+  -You can’t access instance variables from outside the class, only if you create a new "window" into that class.
 
 
 2. What is a block in Ruby?
 
   Your answer:
+  Blocks in Ruby are pieces of code that are surrounded by the curly braces or by do/end. Inside the block, you have certain functions that need to be executed by the code.
 
   Researched answer:
-
+  A block is a piece of code that accepts arguments, and returns a value. A block is always passed to a method call. A block, essentially, is the same thing as a method, except it does not have a name, and does not belong to an object.
 
 
 3. Ruby has an implicit return. What does that mean?
 
   Your answer:
+  In Ruby, we don't use a return the same way we use it in Javascript. In Ruby, an implicit return means that a line inside do/end or a block returns the last value automatically. 
 
   Researched answer:
-
+  Any statement in Ruby returns the value of the last evaluated expression. A simple example is the getter method. An explicit return would be using the return keyword. An implicit return is without the return keyword, this is possible because ruby automatically returns the last evaluated expression. You didn’t explicitly state you want to return, but it still happening, so it implicit.
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
   Your answer:
+  Object-oriented programming is a way of computer programming that uses objects, which are pieces of code that are independent of each other and store data/behavior inside. OOP allows you to make individual objects based off a blueprint in a class.
```

Comment:
Excellent answer regarding OOP. I would also try and say something about functional programming as well as the differences between the two.